reactor-netty-http-brave/src/main/java/reactor/netty/http/client/BraveHttpClientTracing.java

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright (c) 2011-Present VMware, Inc. or its affiliates, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package reactor.netty.http.client;
```

Comment:
The type that we return to the user is `ConnectionObserver`, so that the user can just use `observe(observer)` and thus enable the tracing. But actually it is not just a `ConnectionObserver` but also implements `mapConnect`. The same is for the server - it is not just `ConnectionObserver`, but also implements `mapHandle`. In order to be able to do that I'm using directly Reactor Netty implementation. That's the reason for the package.
Also allows me to do less casting because the `ConnectionObserver` provides a `Connection` while I need HttpClientRequest/HttpServerRequest and HttpClientResponse/HttpServerResponse

Reply:
Let me try something